util/uuid.c

```diff
@@ -1,34 +1,45 @@
-#include <uuid.h>
+#define _DEFAULT_SOURCE
 #include "util/uuid.h"
 
-#if HAS_LIBUUID
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uuid_generate_random(uuid);
-	uuid_unparse(uuid, out);
-	return true;
-}
-#else
-#include <assert.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
 #include <string.h>
-#include <stdlib.h>
+#include <unistd.h>
+
+#ifndef HAS_GETENTROPY
+static int getentropy(void *buffer, size_t len)
+{
+	FILE *urandom;
```

Comment:
Maybe we can make this variable `static`, so that it's opened only once?

That means we'll leak a file on exit, don't know it it's a big deal.